tests/osaka/eip7692_eof_v1/eip7873_tx_create/test_txcreate_validates.py

```diff
@@ -0,0 +1,229 @@
+        initcode_hash: Hash,
+        pre: Alloc,
+        salt: int,
+        evm_code_type: EVMCodeType,
+        value: int,
+        input_size: int,
+    ) -> Tuple[Bytecode, Address | None]:
+        """
+        Return snippet to cause TXCREATE to be called along with an address which
+        will end up in the `compute_eofcreate_address`, or None if that would be the snippet
+        itself.
+        """
+        # Snippet which invokes the TXCREATE itself
+        txcreate_code = Op.TXCREATE(
+            tx_initcode_hash=initcode_hash, salt=salt, value=value, input_size=input_size
+        )
+        # Snippet which resutnrs the TXCREATE result to caller
+        callee_txcreate_code = Op.MSTORE(0, txcreate_code) + Op.RETURN(0, 32)
+        # Snippet which recovers the TXCREATE result from returndata (wipes memory afterwards)
+        returndataload_code = (
+            Op.RETURNDATALOAD
+            if evm_code_type == EVMCodeType.EOF_V1
+            else Op.RETURNDATACOPY(0, 0, Op.RETURNDATASIZE) + Op.MLOAD(0) + Op.MSTORE(0, 0)
+        )
+        match self:
+            case Factory.DIRECT:
+                return txcreate_code, None
+            case Factory.WITH_CALL:
+                callee_address = pre.deploy_contract(callee_txcreate_code)
+                if evm_code_type == EVMCodeType.EOF_V1:
+                    return Op.EXTCALL(address=callee_address) + returndataload_code, callee_address
+                else:
+                    return Op.CALL(address=callee_address) + returndataload_code, callee_address
+            case Factory.WITH_DELEGATECALL:
+                callee_address = pre.deploy_contract(callee_txcreate_code)
+                if evm_code_type == EVMCodeType.EOF_V1:
+                    return Op.EXTDELEGATECALL(address=callee_address) + returndataload_code, None
+                else:
+                    return Op.DELEGATECALL(address=callee_address) + returndataload_code, None
+            case Factory.WITH_STATICCALL:
+                callee_address = pre.deploy_contract(callee_txcreate_code)
+                if evm_code_type == EVMCodeType.EOF_V1:
+                    return Op.EXTSTATICCALL(address=callee_address) + returndataload_code, None
+                else:
+                    return Op.STATICCALL(address=callee_address) + returndataload_code, None
+
+    def __str__(self) -> str:
+        """Return string representation of the enum."""
+        return f"{self.name}"
+
+
+@pytest.mark.with_all_evm_code_types
+# Subset chosen to limit number of test cases
+@pytest.mark.parametrize("code_a", [ValidatedCode.EOFV1_INITCODE, ValidatedCode.LEGACY])
+@pytest.mark.parametrize("code_b", ValidatedCode)
+# Subset chosen to limit number of test cases
+@pytest.mark.parametrize("factory_a", [Factory.DIRECT, Factory.WITH_CALL])
+@pytest.mark.parametrize("factory_b", Factory)
+@pytest.mark.parametrize("value", [0, 1])
+@pytest.mark.parametrize("input_size", [0, 31])
+def test_txcreate_validates(
```

Comment:
I think the amount of parametrization is necessary, I would even add with and without access-list key that targets a storage slot in the newly created address.

Reply:
this seems further away from TXCREATE execution and its validation, but I may be biased knowing the implementation details.

Maybe this could we well enough served by a separate test, which goes more into initcode doing SSTOREs and SLOADs too?

Okay, actually when I think about testing gas charging, it makes total sense to include here, I'll do that, and the above test I'll add to a todo list of tests to add (for EOFCREATE too), in a separate PR